File: src/app/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User } from '../model/user';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http:HttpClient) { }

  getUser(){
    return this.http.get<any>('http://localhost:8080/users/get');
  }

  addUser(user:User){
    return this.http.post<any>('http://localhost:8080/users/addUser',user);
  }

  findByUserId(id:any){
    return this.http.get<any>(`http://localhost:8080/users/getUser/${id}`);
  }

  updateUser(user:User){
    return this.http.put<any>(`http://localhost:8080/users/updateUser`,user);
  }

  deleteUser(id:any){
    return this.http.delete<any>(`http://localhost:8080/users/deleteUser/${id}`)
  }
}

File: BackEnd/slotBooking/src/main/java/com/bosch/slotBooking/services/UserService.java
package com.bosch.slotBooking.services;

import java.util.List;

import com.bosch.slotBooking.model.User;

public interface UserService {
	
	public List<User> getUsers();
	
	public User addUser(User user);
	
	public User findByUserId(String userId);
	
	public User updateUser(User user);
	
	public void deleteByUserId(String userId);

}
